dotnet/src/Extensions/PromptTemplate.Handlebars/HandlebarsPromptTemplate.cs

```diff
@@ -1,60 +1,68 @@
+/// <summary>
+/// Represents a Handlebars prompt template.
+/// </summary>
+internal class HandlebarsPromptTemplate : IPromptTemplate
 {
     /// <summary>
-    /// Constructor for PromptTemplate.
+    /// Default options for built-in Handlebars helpers.
+    /// </summary>
+    /// TODO [@teresaqhoang]: Support override of default options
+    private readonly HandlebarsPromptTemplateOptions _options;
+
+    /// <summary>
+    /// Constructor for Handlebars PromptTemplate.
     /// </summary>
     /// <param name="promptConfig">Prompt template configuration</param>
     /// <param name="loggerFactory">Logger factory</param>
-    public HandlebarsPromptTemplate(PromptTemplateConfig promptConfig, ILoggerFactory? loggerFactory = null)
+    /// <param name="options">Handlebars promnpt template options</param>
+    public HandlebarsPromptTemplate(PromptTemplateConfig promptConfig, ILoggerFactory? loggerFactory = null, HandlebarsPromptTemplateOptions? options = null)
     {
         this._loggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
         this._logger = this._loggerFactory.CreateLogger(typeof(HandlebarsPromptTemplate));
         this._promptModel = promptConfig;
+        this._options = options ?? new();
     }
 
     /// <inheritdoc/>
 #pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
     public async Task<string> RenderAsync(Kernel kernel, KernelArguments? arguments = null, CancellationToken cancellationToken = default)
-#pragma warning restore CS1998
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
     {
-        var handlebars = HandlebarsDotNet.Handlebars.Create();
+        arguments = this.GetVariables(arguments);
+        var handlebarsInstance = HandlebarsDotNet.Handlebars.Create();
 
-        foreach (KernelPlugin plugin in kernel.Plugins)
-        {
-            foreach (KernelFunction function in plugin)
-            {
-                handlebars.RegisterHelper($"{plugin.Name}_{function.Name}", (writer, hcontext, parameters) =>
-                {
-                    var result = function.InvokeAsync(kernel, arguments).GetAwaiter().GetResult();
-                    writer.WriteSafeString(result.ToString());
-                });
-            }
-        }
+        // Add helpers for kernel functions
+        KernelFunctionHelpers.Register(handlebarsInstance, kernel, arguments, this._options.PrefixSeparator, cancellationToken);
 
-        var template = handlebars.Compile(this._promptModel.Template);
+        // Add built-in system helpers
+        KernelSystemHelpers.Register(handlebarsInstance, arguments, this._options);
 
-        return template(this.GetVariables(arguments));
+        // Register any custom helpers
+        this._options.RegisterCustomHelpers?.Invoke(handlebarsInstance, arguments);
+
+        var template = handlebarsInstance.Compile(this._promptModel.Template);
```

Comment:
We have to do this "Compile" per RenderAsync call because the compilation depends on functions that have been registered with it and those can change?

Reply:
Yep exactly, we spin up a new Hb instance per prompt template so the functions should change on each render